### PR TITLE
Periodic bonds, unit cell tests, IRMOF-1 tests.

### PR DESCRIPTION
**Pull request by kbsezginel**
- Periodic bond extrapolation method and tests.
- IRMOF-1 tests.
- Extend unit cell method now returns a new MOF object.
- Tests for calculating unit cell vectors for different crystal systems.
- Extrapolate bonds now checks z-distance before calculating the actual distance.
- PEP8 formatting for tests.

### CODE REVIEW

File: lammps_data/crystal.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import math
 import os
+import copy
```
Comment (paulboone) [inline lammps_data/crystal.py:3]: good find! I didn't know about this one. I've been doing it by hand. :\

File: tests/test_unit_cell_vectors.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+from lammps_data.crystal import Packing
+
+
+def test_cubic_unit_cell_vectors():
```
Comment (paulboone) [inline tests/test_unit_cell_vectors.py]: _should_what?

File: lammps_data/bonds.py [hunk 3]
```diff
@@ -1,50 +1,51 @@
...
+    2.32, 1.96,
+    1.80, 1.63, 1.76, 1.74, 1.73, 1.72, 1.68,
+    1.69, 1.68, 1.67, 1.66, 1.65, 1.64, 1.70,
+    1.62, 1.52, 1.46, 1.37, 1.31, 1.29, 1.22, 1.23, 1.24, 1.33,
+    1.44, 1.44, 1.51, 1.45, 1.47, 1.42,
+    2.23, 2.01,
+    1.86, 1.75, 1.69, 1.70, 1.71, 1.72, 1.66,
+    1.66, 1.68, 1.68, 1.65, 1.67, 1.73, 1.76,
+    1.61, 1.57, 1.49, 1.43, 1.41, 1.34, 1.29, 1.28, 1.21, 1.22,
+    1.36, 1.43, 1.62, 1.75, 1.65, 1.57
```
Comment (paulboone) [inline lammps_data/bonds.py:22]: ought to separate whitespace into a separate commit so it is clearer what the logical change is

File: lammps_data/bonds.py [hunk 4]
```diff
@@ -1,50 +1,51 @@
...
-                        (atoms_z[j][1][1] - atoms_z[i][1][1])**2 +
-                        (atoms_z[j][1][2] - atoms_z[i][1][2])**2 ) ** 0.5
-
             if abs(atoms_z[j][1][2] - atoms_z[i][1][2]) > max_cutoff:
                 break
 
+            distance = ((atoms_z[j][1][0] - atoms_z[i][1][0])**2 +
+                        (atoms_z[j][1][1] - atoms_z[i][1][1])**2 +
+                        (atoms_z[j][1][2] - atoms_z[i][1][2])**2) ** 0.5
+
```
Comment (paulboone) [inline lammps_data/bonds.py:46]: Yep, this makes sense. Also, this sort of thing is exactly why the tests are helpful... you ran the tests immediately afterward and were confident that nothing broke?

File: tests/test_extrapolate_periodic_bonds.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def test_periodic_bond_tuples_should_be_sorted_by_atom_index():
+    cell_vectors = [[10, 0, 0], [0, 10, 0], [0, 0, 10]]
+    atoms = [(0.0, 0.0, 9.9, 1), (0.0, 0.0, 10.06, 1)]
+    bonds = extrapolate_periodic_bonds(atoms, cell_vectors)
+    assert bonds == [(0, 1)]
+    atoms = [(0.0, 0.0, 10.06, 1), (0.0, 0.0, 9.9, 1)]
+    bonds = extrapolate_periodic_bonds(atoms, cell_vectors)
+    assert bonds == [(0, 1)]
```
Comment (paulboone) [inline tests/test_extrapolate_periodic_bonds.py:63]: I like this separated out like this because it is really clear the distinction between periodic / nonperiodic tests.

Only problem is that if we add tests to one or the other, we now need to manually duplicate them. 

should we consider parameterized fixtures?

File: tests/test_irmof1.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+from lammps_data.crystal import MOF
+from lammps_data.bonds import extrapolate_bonds, extrapolate_periodic_bonds
+from lammps_data.angles import get_angles
+from lammps_data.dihedrals import get_dihedrals
+
+expected_cell_vectors = [[25.832, 0, 0], [0, 25.832, 0], [0, 0, 25.832]]
+
+
+def test_irmof1():
```
Comment (paulboone) [inline tests/test_irmof1.py]: should_what?

**Review by paulboone — CHANGES_REQUESTED** [review]
Looks good, mostly just test names and some things to think about.

File: lammps_data/bonds.py (hunk 4 above)
Comment (kbsezginel) [reply lammps_data/bonds.py:46]: Yes, this was when I really appreciated we had tests!

**Comment by kbsezginel** [comment]
Renamed tests so they should be more clear now.

**Comment by kbsezginel** [comment]
Renamed tests so they should be more clear now.

**Comment by kbsezginel** [comment]
Renamed tests so they should be more clear now.

**Review by paulboone — APPROVED** [review]
